src/test/fuzz/minisketch.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <minisketch.h>
+#include <minisketchwrapper.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/check.h>
+
+#include <map>
+#include <numeric>
+
+FUZZ_TARGET(minisketch)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    const auto capacity{fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, 200)};
+    Minisketch sketch_a{Assert(MakeMinisketch32(capacity))};
+    Minisketch sketch_b{Assert(MakeMinisketch32(capacity))};
+
+    // Fill two sets and keep the difference in a map
+    std::map<uint32_t, bool> diff;
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 10000)
+    {
+        const auto entry{fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, std::numeric_limits<uint32_t>::max() - 1)};
+        const auto KeepDiff{[&] {
+            bool& mut{diff[entry]};
+            mut = !mut;
+        }};
+        CallOneOf(
+            fuzzed_data_provider,
+            [&] {
+                sketch_a.Add(entry);
+                KeepDiff();
+            },
+            [&] {
+                sketch_b.Add(entry);
+                KeepDiff();
+            },
+            [&] {
+                sketch_a.Add(entry);
+                sketch_b.Add(entry);
+            });
+    }
+    const auto num_diff{
+        std::accumulate(diff.begin(), diff.end(), size_t{0}, [](size_t n, const std::pair<const uint32_t, bool>& e) { return n + e.second; })};
+
+    Minisketch sketch_ar{MakeMinisketch32(capacity)};
+    Minisketch sketch_br{MakeMinisketch32(capacity)};
+    sketch_ar.Deserialize(sketch_a.Serialize());
+    sketch_br.Deserialize(sketch_b.Serialize());
+
+    Minisketch sketch_diff{std::move(fuzzed_data_provider.ConsumeBool() ? sketch_a : sketch_ar)};
+    sketch_diff.Merge(fuzzed_data_provider.ConsumeBool() ? sketch_b : sketch_br);
+
+    if (capacity >= num_diff) {
```

Comment:
We'll probably only hit this condition ~3% of the time, right (very roughly)?

Reply:
Am I correct in my understanding of Minisketch, that if `if (capacity < num_diff)` then `ssize_t num_differences = minisketch_decode(sketch_diff, capacity, differences);` will return `<0` indicating that there are more than `capacity` differences in the set? Can we add an `else` clause and assert this?

Reply:
> We'll probably only hit this condition ~3% of the time, right (very roughly)?

That depends on the fuzz engine's randomness and exploration algorithm.

Reply:
> will return <0

I had the same thinking, but it may still succeed "accidentally" on some inputs.

Reply:
Yep that makes sense. That's where I arrived after reading the Minisketch README more, but wanted to double check anyways